wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceRuntimes.java

```diff
@@ -470,7 +475,8 @@ void recoverOne(RuntimeInfrastructure infra, RuntimeIdentity identity) throws Se
 
     InternalRuntime runtime;
     try {
-      InternalEnvironment internalEnvironment = infra.estimate(environment);
+      InternalEnvironment internalEnvironment =
```

Comment:
Looks like previously created `InternalEnvironment` should be used instead of the creation of the new one.

Reply:
Not sure I understand it, could you please rephrase

Reply:
Nevermind - I missed lines collapsed by Github and misunderstood your changes because of that. Sorry